pages/blog/miniseries-part_1.md

```diff
@@ -0,0 +1,234 @@
+
+Initially, I used OpenAPI to describe the API and the great thing was their tooling allowed me to generate clients and servers in different languages which speed up the implementation process.
+
+I soon encountered a use case that required me to push data to the game server, and solving this with REST was possible, but cumbersome. So I started exploring different alternatives, in terms of event-driven architecture, however, none could be described using OpenAPI removing tooling, so I had to find alternatives. 
+
+That was when I vaguely remembered a meeting in the company where AsyncAPI was mentioned. Around that time, at work, we began to switch from a custom socket protocol to [NATS](https://nats.io/), and at the same time they spend some time figuring out how to mainstream the process and document the APIs, and this is where they found and adopted AsyncAPI. So I started to use AsyncAPI in my project, which sparked my first ever contribution to an open-source project, but that is a story for another time, maybe.
+
+So this blog post is a dedication to that experience, showcasing how I use AsyncAPI to document and generate code to speed up the development process, and maybe spark your interest in helping us build the best tooling possible.
+
+# To that end
+
+Explaining something is always better with actual examples, therefore I will be creating a little system to show you how code generation can help the development process. 
+<figure>
+  <img src="/img/posts/jonaslagoni-miniseries-part1/blog-miniseries-general-setup.webp" title="General setup of the project"/>
+  <figcaption className="text-center text-gray-400 text-sm">The general setup of the project, with the two applications game server and processor. The round dot between "some broker" and the applications represent how others may grab/interact with the application, ergo its API.</figcaption>
+</figure>
+
+I will be creating a system of two applications, a **game server** and a **processor** using a micro-service architecture with no public-facing API. How a player interacts with the **game server** could be through a phone, a computer, Xbox, or PlayStation, for us this is out of scope for this project. For this blog post I only care about the interaction between the **game server** and the **processor**. 
+
+The **game server** will produce the following events: when players join the server, pick up items in-game, uses the chat, hit one another, and eventually disconnect. It will be implemented to simulate players at random intervals is joining the server, picking up some items, etc, and eventually disconnecting, to provide a sense of realism. 
```

Comment:
```suggestion
The **game server** will produce the following events: when players join the server, pick up items in-game, uses the chat, hit one another, and eventually disconnect. It will be implemented to simulate players at random intervals is joining the server, picking up some items, etc, and eventually disconnecting to provide a sense of realism. 
```